addon/components/flexberry-dropdown.js

```diff
@@ -165,12 +165,18 @@ export default FlexberryBaseComponent.extend({
       Ember.Logger.error(`Wrong type of flexberry-dropdown \`items\` propery: actual type is ${Ember.typeOf(items)}, but array is expected.`);
     }
 
-    var value = this.get('value') || null;
-    if (needChecksOnValue && !Ember.isNone(value) && items.indexOf(value) < 0) {
+    // Convert 'value' and 'items' to strings because flexberry-dropdown interpret selected value as string commonly.
+    let value = this.get('value') || null;
+    let stringValue = value ? value.toString() : null;
+    let itemsString = items.map(function(item) {
+      return item ? item.toString() : null;
+    });
+
+    if (needChecksOnValue && !Ember.isNone(stringValue) && itemsString.indexOf(stringValue) < 0) {
```

Comment:
It was investigated by me that items' values are kept as 'data-value' attributes at semantic ui dropdown. So if the user selects new value component returns strings.
But when value is set initially it can be not string.
That's why it is usefully to convert everything to string before check to my mind.